prow/tide/tide.go

```diff
@@ -1824,3 +1850,54 @@ func nonFailedBatchByNameBaseAndPullsIndexFunc(obj runtime.Object) []string {
 
 	return []string{nonFailedBatchByNameBaseAndPullsIndexKey(pj.Spec.Job, pj.Spec.Refs)}
 }
+
+type failureCommenter interface {
+	HasMergeFailureComment(*PullRequest) (bool, error)
+	CreateMergeFailureComment(*PullRequest, error) error
+}
+
+type mergeFailureCommenter struct {
+	ghc     githubClient
+	botName string
+}
+
+func newFailureCommenter(ghc githubClient) (*mergeFailureCommenter, error) {
+	botName, err := ghc.BotName()
+	if err != nil {
+		return nil, err
+	}
+	return &mergeFailureCommenter{
+		ghc:     ghc,
+		botName: botName,
+	}, nil
+}
+
+func (a *mergeFailureCommenter) HasMergeFailureComment(pr *PullRequest) (bool, error) {
+	comments, err := a.ghc.ListIssueComments(
+		string(pr.Repository.Owner.Login),
+		string(pr.Repository.Name),
+		int(pr.Number),
+	)
+	if err != nil {
+		return false, err
+	}
+	for _, comment := range comments {
+		if comment.User.Login == a.botName && strings.Contains(comment.Body, mergeFailureComment) {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (a *mergeFailureCommenter) CreateMergeFailureComment(pr *PullRequest, err error) error {
+	message := mergeFailureComment
+	if err != nil {
+		message = fmt.Sprintf("%s The following are the error details:\n```\n%s\n```\n", mergeFailureComment, err)
```

Comment:
I vaguely remember this markdown formatting sort of thing not working as expected when posting via the API - can we use `<code>`?

Reply:
Sorry for the delay, I missed a few updates from this repo. All reviews are now addressed.